src/main/scala-2.11/markets/clearing/engines/CDAMatchingEngine.scala

```diff
@@ -84,28 +83,28 @@ class CDAMatchingEngine(askOrdering: PriceOrdering[AskOrder],
         val price = formPrice(incoming, askOrder)
 
         if (residualQuantity < 0) {  // incoming order is smaller than existing order
-          val (filledOrder, residualOrder) = askOrder.split(-residualQuantity)
-          val totalMatch = TotalMatch(filledOrder, incoming, price)
-          _askOrderBook.add(residualOrder)  // SIDE EFFECT!
-          matches.enqueue(totalMatch)
+          val (_, residualAskOrder) = askOrder.split(-residualQuantity)
+          val matching = Matching(askOrder, incoming, Some(price), Some(residualAskOrder), None)
+          _askOrderBook.add(residualAskOrder)  // SIDE EFFECT!
+          matchings.enqueue(matching)
         } else if (residualQuantity == 0) {  // no rationing for incoming order!
-          val totalMatch = TotalMatch(askOrder, incoming, price)
-          matches.enqueue(totalMatch)
+          val matching = Matching(askOrder, incoming, Some(price), None, None)
+          matchings.enqueue(matching)
         } else {  // incoming order is larger than existing order and will be rationed!
-          val (filledOrder, residualOrder) = incoming.split(residualQuantity)
-          val partialMatch = PartialMatch(askOrder, filledOrder, price)
-          accumulateAskOrders(residualOrder, matches.enqueue(partialMatch))
+          val (_, residualBidOrder) = incoming.split(residualQuantity)
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Field name does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/davidrpugh/markets-sandbox/file/1545342681/issues/source?bid=2921361&fileBranchId=3015559#l94)